admin/src/templates/modal.html

```diff
@@ -6,9 +6,9 @@ <h2 translate translate-values="titleParams">{{titleKey}}</h2>
   <div class="modal-body" ng-transclude></div>
   <div class="modal-footer">
     <p class="alert text-left" ng-class="status.severity === 'warning' ? 'alert-warning' : 'alert-danger'" ng-if="status.error" translate>{{status.error}}</p>
-    <a class="btn cancel" ng-class="{ 'disabled': status.processing }" ng-click="onCancel()" translate>{{cancelKey || 'Cancel'}}</a>
+    <a class="btn cancel" ng-attr-id="{{ id + '-cancel-btn' }}" ng-class="{ 'disabled': status.processing }" ng-click="onCancel()" translate>{{cancelKey || 'Cancel'}}</a>
     <a class="btn btn-danger pull-left" ng-show="showDelete" ng-click="onDelete()"><i class="fa fa-trash-o"></i>&nbsp;<span translate>{{deleteKey || 'Delete'}}</span></a>
-    <a class="btn submit" ng-if="submitKey" ng-class="{ 'btn-primary': !danger, 'btn-danger': danger, 'disabled': disableSubmit }" mm-enter="onSubmit()" ng-click="onSubmit()" ng-show="!status.processing" translate>{{submitKey}}</a>
+    <a class="btn submit" ng-attr-id="{{ id + '-submit-btn' }}" ng-if="submitKey" ng-class="{ 'btn-primary': !danger, 'btn-danger': danger, 'disabled': disableSubmit }" mm-enter="onSubmit()" ng-click="onSubmit()" ng-show="!status.processing" translate>{{submitKey}}</a>
```

Comment:
Nice work here but it's going to need a bit of change. 

For attributes or ids we add for testing we need to use a prefix  of `test-`  for the key.  We are trying to avoid any collisions our test attributes might cause if the code actually needs to use the id or any other common attribute. 

See https://github.com/medic/cht-core/blob/master/TESTING.md#adding-identifiers for some additional details.

Reply:
Done